### DISCUSSION TITLE
Database configuration as URL/DSN with timezone

### DISCUSSION THREAD

**Discussion by Gounlaf** [laravel/framework#33336] [General]
- Laravel Version: 6.18.22
- PHP Version: 7.4.7
- Database Driver & Version: `pdo_mysql`

### Description:

I want to configure the timezone of a database connection. It doesn't when using a DSN.

### Steps To Reproduce:

* Declare a connection using the "url" options instead of splitting all part manually; `mysql://user:pwd@host:port/database_name?timezone=%2B00%3A00` or `mysql://user:pwd@host:port/database_name?timezone=+00:00`
* Make a request via Eloquent
* You will get an error about the timezone: it's parsed/decoded as ` 00:00` instead of `+00:00`

Encoded (`%2B00%3A00`) or "raw" version (`+00:00`) doesn't work
Things happens in ConfigurationUrlParser class, when options (queryString) are parsed.

Double encoding the value `%252B00%253A00` works, but I think it's more a hack than the real way to configure the URI/DSN.

Regards.

**Comment by Gounlaf** [comment]
@driesvints Hi,

You transfered the issue as a question ; but I'm reporting a bug, not a question...

**Comment by Gounlaf** [comment]
Theses 2 tests failed

Edit: diff moved in branch https://github.com/Gounlaf/laravel-framework/commit/32bbc2acbbcb74ac1d176596a6daa2c56ad937b1

**Comment by driesvints** [comment]
@Gounlaf feel free to PR a fix with a failing test if you believe something is broken. Thanks
